src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EndpointInfoSourceTests.cs

```diff
@@ -269,15 +276,20 @@ public async Task<EndpointInfo> WaitForNewEndpointInfoAsync(AppRunner runner, Ti
 
                 _outputHelper.WriteLine($"[Wait] Wait for App{runner.AppId} notification");
                 timeoutCancellation.CancelAfter(timeout);
-                EndpointInfo endpointInfo = await addedEndpointInfoSource.Task;
+                IEndpointInfo endpointInfo = await addedEndpointInfoSource.Task;
                 _outputHelper.WriteLine($"[Wait] Received App{runner.AppId} notification");
 
                 return endpointInfo;
             }
 
-            internal override void OnAddedEndpointInfo(EndpointInfo info)
+            public Task OnBeforeResumeAsync(IEndpointInfo endpointInfo, CancellationToken cancellationToken)
+            {
+                return Task.CompletedTask;
+            }
+
+            public void OnAddedEndpointInfo(IEndpointInfo info)
             {
-                _outputHelper.WriteLine($"[Source] Added: {info.DebuggerDisplay}");
+                _outputHelper.WriteLine($"[Source] Added: {info.GetDebuggerDisplay()}");
                 
                 lock (_addedEndpointInfoSources)
                 {
```

Comment:
I realize this is out of the scope of this PR but doesn't this (lines 298 and 308) throw an exception from modifying a collection while iterating it?
```cs
foreach (var sourceTuple in _addedEndpointInfoSources)
{
     _addedEndpointInfoSources.Remove(sourceTuple);
}
```

Reply:
Yup, it should throw. It probably doesn't because (my guess is that) it happens to remove the last item in the list, so there isn't anything else to iterate. I'll fix it in this PR.